CI: 17 dub and meson project fails in OpenSUSE container

The CI image builder is failing for OpenSUSE: https://github.com/mesonbuild/meson/actions/runs/24584013531/job/71888918845?pr=15711#step:5:643

In a Fedora rawhide container, it fails the same way with latest dmd+dub:
```shell
$ curl -fsS https://dlang.org/install.sh | bash -s dmd
$ ./run_single_test.py "./test cases/d/17 dub and meson project"/
WARNING: No include directory found parsing "gcc -xobjective-c -E -v -" output

host machine compilers

c      : [gcc]      cc (gcc 16.0.1 "cc (GCC) 16.0.1 20260321 (Red Hat 16.0.1-0)")
cpp    : [not found]
cs     : [not found]
cuda   : [not found]
cython : [not found]
d      : [dmd]      dmd (dmd 2.112.0 "DMD64 D Compiler v2.112.0")
fortran: [not found]
java   : [not found]
linearasm: [not found]
masm   : [not found]
nasm   : [not found]
objc   : [not found]
objcpp : [not found]
rust   : [not found]
swift  : [not found]
vala   : [not found]

tools

ninja  : /usr/bin/ninja (1.13.2)
cmake  : not found
hotdoc : not found

FAIL: d: 17 dub and meson project
[...]
    raise DependencyException('Multiple version requirements are not solvable for raw dub dependencies')
mesonbuild.dependencies.base.DependencyException: Dependency lookup for urld with method 'dub' failed: Multiple version requirements are not solvable for raw dub dependencies
```

If I downgrade to dub-1.40.0, it works (fails as expected because I haven't got `urld` in this container):
```shell
$ curl -fsS https://dlang.org/install.sh | bash -s dub-1.41.0
$ source ~/dlang/dub-1.40.0/activate
$ ./run_single_test.py "./test cases/d/17 dub and meson project"/
[...]
Found DUB: /root/dlang/dub-1.40.0/dub (version 1.40.0)
Determining dependency 'urld' with DUB executable '/root/dlang/dub-1.40.0/dub'
Searching in DUB cache for compatible liburld.a
ERROR: urld@3.0.0 not found
You may try the following command to install the necessary DUB libraries:
dub build --deep --arch=x86_64 --compiler=gdc --build=debug
Run-time dependency urld for d found: NO  (tried dub)

test cases/d/17 dub and meson project/meson.build:14:7: ERROR: Dependency "urld" not found (tried dub)
```

With bad dub (1.41.0):
```
(dmd-2.112.0)[root@cd5d5aa90c81 17 dub and meson project]# dub
core.exception.AssertError@source/dub/project.d(197): Non-optional dependency '17-dub-meson-project:multi-configuration' of '17-dub-meson-project' not found in dependency tree!?.
----------------
??:? [0x561ff790a3a6]
??:? [0x561ff790a022]
??:? [0x561ff7932296]
??:? [0x561ff791277c]
??:? [0x561ff7909082]
??:? [0x561ff75d99f0]
??:? [0x561ff75d9784]
??:? [0x561ff75e48af]
??:? [0x561ff75da1cd]
??:? [0x561ff75da0a5]
??:? [0x561ff78a2ac5]
??:? [0x561ff78a3bfd]
??:? [0x561ff789baf6]
??:? [0x561ff791244c]
??:? [0x561ff7912346]
??:? [0x561ff791219c]
??:? [0x7f175bbd7680]
??:? __libc_start_main [0x7f175bbd7797]
??:? [0x561ff75b3c49]
```

vs good dub (1.40.0):
```
(dub-1.40.0)[root@cd5d5aa90c81 17 dub and meson project]# dub
    Starting Performing "debug" build using gdc for x86_64.
    Building 17-dub-meson-project:multi-configuration ~master: building configuration [lib]
    Building dubtestproject:test3 1.2.0: building configuration [library]
    Building urld 3.0.0: building configuration [library]
    Building 17-dub-meson-project ~master: building configuration [application]
/usr/bin/ld.bfd: /root/.dub/cache/dubtestproject/1.2.0/+test3/build/library-debug-omUtOqLIurldy_utNcHajw/libtest3.a: in function `D main':
/root/.dub/packages/dubtestproject/1.2.0/dubtestproject/source/test3/test.d:3: multiple definition of `D main'; /tmp/ccT9g98B.o:/tmp/meson/test cases/d/17 dub and meson project/source/app.d:1: first defined here
/usr/bin/ld.bfd: /root/.dub/cache/dubtestproject/1.2.0/+test3/build/library-debug-omUtOqLIurldy_utNcHajw/libtest3.a: in function `main':
/usr/lib/gcc/x86_64-redhat-linux/16/include/d/core/internal/entrypoint.d:27: multiple definition of `main'; /tmp/ccT9g98B.o:/usr/lib/gcc/x86_64-redhat-linux/16/include/d/core/internal/entrypoint.d:27: first defined here
collect2: error: ld returned 1 exit status
Error gdc failed with exit code 1.
```

--- a/docs/markdown/Dependencies.md
+++ b/docs/markdown/Dependencies.md
@@ -868,7 +868,7 @@ $ wx-config --libs std stc
 ## Zlib
 
 Zlib ships with pkg-config and cmake support, but on some operating
-systems (windows, macOs, FreeBSD, dragonflybsd, android), it is provided as
+systems (windows, macOs, FreeBSD, DragonFly BSD, android), it is provided as
 part of the base operating system without pkg-config support. The new
 System finder can be used on these OSes to link with the bundled
 version.

--- a/docs/markdown/Dependencies.md
+++ b/docs/markdown/Dependencies.md
@@ -868,7 +868,7 @@ $ wx-config --libs std stc
 ## Zlib
 
 Zlib ships with pkg-config and cmake support, but on some operating
-systems (windows, macOs, FreeBSD, DragonFly BSD, android), it is provided as
+systems (Windows, macOS, FreeBSD, DragonFly BSD, Android), it is provided as
 part of the base operating system without pkg-config support. The new
 System finder can be used on these OSes to link with the bundled
 version.

--- a/docs/markdown/Release-notes-for-0.64.0.md
+++ b/docs/markdown/Release-notes-for-0.64.0.md
@@ -94,7 +94,7 @@ method.
 ## BSD support for the `jni` dependency
 
 This system dependency now supports all BSD systems that Meson currently
-supports, including FreeBSD, NetBSD, OpenBSD, and DragonflyBSD.
+supports, including FreeBSD, NetBSD, OpenBSD, and DragonFly BSD.
 
 ## Credentials from `~/.netrc` for `https` URLs
 

--- a/mesonbuild/dependencies/dub.py
+++ b/mesonbuild/dependencies/dub.py
@@ -351,6 +351,8 @@ class DubDependency(ExternalDependency):
         ret, res, err = self._call_dubbin(describe_cmd)
         if ret == 0:
             return (json.loads(res), helper_build, source)
+        else:
+            mlog.debug('DUB describe (raw) failed: ' + err)
 
         pack_spec = self.name
         if self.version_reqs is not None:

--- a/mesonbuild/envconfig.py
+++ b/mesonbuild/envconfig.py
@@ -363,7 +363,7 @@ class MachineInfo(HoldableObject):
         return self.system == 'openbsd'
 
     def is_dragonflybsd(self) -> bool:
-        """Machine is DragonflyBSD?"""
+        """Machine is DragonFly BSD?"""
         return self.system == 'dragonfly'
 
     def is_freebsd(self) -> bool:

--- a/test cases/common/221 zlib/meson.build
+++ b/test cases/common/221 zlib/meson.build
@@ -1,7 +1,7 @@
 project('zlib system dependency', 'c')
 
 if not ['darwin', 'freebsd', 'dragonfly', 'windows', 'android'].contains(host_machine.system())
-  error('MESON_SKIP_TEST only applicable on macOS, FreeBSD, DragonflyBSD, Windows, and Android.')
+  error('MESON_SKIP_TEST only applicable on macOS, FreeBSD, DragonFly BSD, Windows, and Android.')
 endif
 
 cc = meson.get_compiler('c')

--- a/test cases/d/17 dub and meson project/multi-configuration/dub.json
+++ b/test cases/d/17 dub and meson project/multi-configuration/dub.json
@@ -1,14 +1,16 @@
 {
   "name": "multi-configuration",
-  "configurations": {
-    "app": {
+  "configurations": [
+    {
+      "name": "app",
       "targetType": "executable"
     },
-    "lib": {
+    {
+      "name": "lib",
       "targetType": "library",
       "excludedSourceFiles": [
         "source/app.d"
       ]
     }
-  }
+  ]
 }

--- a/unittests/allplatformstests.py
+++ b/unittests/allplatformstests.py
@@ -1752,7 +1752,7 @@ class AllPlatformTests(BasePlatformTests):
             rpath = get_rpath(os.path.join(self.builddir, each))
             self.assertTrue(rpath, f'Rpath could not be determined for {each}.')
             if is_dragonflybsd():
-                # DragonflyBSD will prepend /usr/lib/gccVERSION to the rpath,
+                # DragonFly BSD will prepend /usr/lib/gccVERSION to the rpath,
                 # so ignore that.
                 self.assertTrue(rpath.startswith('/usr/lib/gcc'))
                 rpaths = rpath.split(':')[1:]

